### DISCUSSION TITLE
Advice running on Raspberry Pi

### DISCUSSION THREAD

**Discussion by jpmiller25** [immich-app/immich#783] [Q&A] [answered]
Hi, I've been running this for about a month on a RPi4 - 4Gb running Ubuntu 22.04 64 bit. I'm loving it and I'm very excited to see where this project goes.

I've had 2 issues so far, one is that backup from iphone continually reverts back to X amount of photos to upload and never finishes. I'm hopeful PR 595 fixes that.

The second is sometimes the CLI tool is freezing my system when uploading large numbers of files (only 1800, I've seen others uploading lots more with that tool!) I'll try the new --threads feature, maybe setting to 1 will help.

I'd like some pointers on getting the best performance possible out of my setup. I have my photos and database on a 4TB HDD connected to USB3.0. Would it make any difference in speed if I put the database on the SD card? Or would a faster SSD be better? Or is my memory the limiting factor here, and faster r/w speed wouldn't make a difference with such small memory?

Would it make any sense to disable machine learning features and is there a way to do that yet?

Thanks for all the work going into this project!! I hope to help however I can!

**Comment by alextran1502** [comment, marked as answer]
Hello @jpmiller25,

the X amount of photos you see that are not uploaded are the duplicated one, therefore it is rejected by the server. We haven't put in a mechanism to filter those duplicated files yet. I am hoping to implement it in the next couple of releases.

The CLI has reported issue of eating all the RAM when reading and uploading large video files. You can try the thread option. I am working on a different version of the CLI to better restructure that project (It starts as a quick workaround to get things up to Immich and has grown into a more popular tool) and to improve performance :)

For the database, putting it on the SD Card is fine but make sure to perform the backup frequently because the SD Card can failed faster than the HDD. The amount of database transaction in the app is not much, even with a million assets, that is still quite minimal for database workload.
